fe/fe-core/src/main/java/org/apache/doris/rewrite/CompoundPredicateWriteRule.java

```diff
@@ -0,0 +1,102 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.rewrite;
+
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.BoolLiteral;
+import org.apache.doris.analysis.CompoundPredicate;
+import org.apache.doris.analysis.Expr;
+import org.apache.doris.common.AnalysisException;
+
+import java.util.List;
+
+/**
+ * add Rewrite CompoundPredicates 'OR' 'AND' 'NOT' Rule
+ * It can be applied to pre-analysis expr trees and therefore does not reanalyze
+ * the transformation output itself.
+ * Examples:
+ * OR:
+ * (-2==2 OR city_id=2) ==> city_id=2
+ * (city_id=2 OR -2==2) ==> city_id=2
+ * -5!=-5 OR citycode=0 ==> citycode=0
+ * AND:
+ * (citycode=0 AND 1=1) ==> citycode=0
+ * -5=-5 AND citycode=0 AND 2=2 ==> citycode=0
+ */
+
+public class CompoundPredicateWriteRule implements ExprRewriteRule {
+    public static ExprRewriteRule INSTANCE = new CompoundPredicateWriteRule();
+
+    @Override
+    public Expr apply(Expr expr, Analyzer analyzer) throws AnalysisException {
+
+        if (!(expr instanceof CompoundPredicate)) return expr;
+        CompoundPredicate cp = (CompoundPredicate) expr;
+
+        List<Expr> children = cp.getChildren();
+        if (children.size() != 2) {
+            return expr;
+        }
+
+        /*
+         *  'OR' 'AND' rewrite rule
+         *  case true AND expr ==> expr
```

Comment:
Typo. 
It's better to unify the style, a.k.a. use uppercase or lowercase for all the keywords `and` and `or`, rather than uppercase `AND` and camelcase `Or`. Maybe just use lowercase `and` and `or` is enough.

Reply:
ok，i will change those and commit，thanks for your code review，i will pay attention to these details staring today.